spec/rspec/core/world_spec.rb

```diff
@@ -224,6 +224,15 @@ module RSpec::Core
             world.announce_filters
           end
         end
+
+        context "with a filter but with silence_filter_announcements" do
+          it "does not announce" do
+            allow(configuration).to receive(:silence_filter_announcements?) { true }
```

Comment:
Good thought. I just made the change. I did it as a new commit to show the progression but feel free to squash it into the first one later on.